weave

```diff
@@ -805,6 +811,18 @@ proxy_args() {
     done
 }
 
+proxy_addr() {
+    if addr=$(docker logs weaveproxy | head -n3 | grep -oE "proxy listening on .*"); then
```

Comment:
I had suggested `grep -m 1`, which would make this resilient to extra lines being. Does that not have the desired effect?